### ISSUE TITLE
Add a hook for retrieving ActiveRecord fragments for included relationships

### ISSUE THREAD

**Issue by nihalgonsalves** [JSONAPI-Resources/jsonapi-resources#958]
For example, if you have a resource `UserResource` with a method like this

```ruby
def friends
   @model.friends.size
end
```

It will load `user.friends` separately for each model in the list. So you can optimise the query like this:

```ruby
class self.records
    _model_class.all.includes(:friends)
end
```

Then when requesting `GET /users/` it now works well. However, say if you request something like `GET /groups/?include=users`, the `records` method on `UserResource` is never called, so there is no query optimisation. 

Instead of overriding `self.apply_includes(records, options = {})` on each resource that has a relationship to `UserResource`, it would be nice to have a way to do this only once per resource.

@lgebhardt as discussed on Gitter
